webapi/src/main/twirl/org/knora/webapi/queries/sparql/v2/isPropertyUsed.scala.txt

```diff
@@ -0,0 +1,53 @@
+@*
+ * Copyright © 2015-2021 the contributors (see Contributors.md).
+ *
+ * This file is part of Knora.
+ *
+ * Knora is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Knora is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with Knora.  If not, see <http://www.gnu.org/licenses/>.
+ *@
+
+@import org.knora.webapi.IRI
+
+@*
+ * Checks whether a property is used (i.e., is the property of any statements).
+ *
+ * @param triplestore the name of the triplestore being used.
+ * @param propertyIri the IRI of the property.
+ * @param ignoreKnoraConstraints if true, knora-base:subjectClassConstraint and knora-base:objectClassConstraint will be ignored.
+ * This is necessary when modifying the cardinalities of a class.
+ * @param ignoreRdfSubjectAndObject if true, rdf:subject and rdf:object will be ignored. This is necessary when checking
+ * for references to a resource that is to be erased.
+ *@
+@(triplestore: String,
+  internalPropertyIri: IRI,
+  ignoreKnoraConstraints: Boolean = false,
+  ignoreRdfSubjectAndObject: Boolean = false)
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+
+SELECT DISTINCT ?isUsed
+@* Ensure that inference is not used in this query. *@
+@if(triplestore.startsWith("graphdb")) {
+    FROM <http://www.ontotext.com/explicit>
+}
+WHERE {
+    BIND(IRI("@internalPropertyIri") AS ?property)
```

Comment:
1. This will always be true because it checks any statement from ontology and data, and since the property is used by the class that you want to change, it will always be true. This is the same if you use isEntityUsed template, that will also be always true.
2.  If you wish to make a query that returns boolean, use ASK instead of SELECT.

Reply:
Thanks, will change that.

Reply:
done